src/cloudai/workloads/aiconfig/predictor.py

```diff
@@ -0,0 +1,303 @@
+# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+# Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import sys
+from typing import Any, Dict, Optional, cast
+
+
+def _to_enum(enum_cls: Any, value_or_name: Any) -> Any:
+    """
+    Convert a string or enum value to the corresponding enum instance.
+
+    If value_or_name is already an enum instance, return it as-is.
+    """
+    if isinstance(value_or_name, enum_cls):
+        return value_or_name
+    if isinstance(value_or_name, str):
+        return enum_cls[value_or_name]
+    return enum_cls(value_or_name)
+
+
+def _validate_nextn(nextn: int, nextn_accept_rates: Optional[list[float]]) -> list[float]:
+    if nextn > 0 and nextn_accept_rates is None:
+        raise ValueError("nextn_accept_rates must be provided when nextn > 0")
+    return nextn_accept_rates or []
+
+
+def _ensure_aiconfigurator_available(*, need_inference_session: bool) -> dict[str, Any]:
```

Comment:
1. All imports should be done directly on top of the file.
2. This approach with classes/functions returned as dict completely ruins type checking for this file, all items of the dict are marked as `Any`.